File: 128-longest-consecutive-sequence/128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        if len(nums) <= 1: return len(nums)
        
        hash_map = {}
        for num in nums:
            hash_map[num] = None
            
        longest = 0
        for num in nums:
            if hash_map[num] != None: continue
            
            cnt = 1
            cur = num-1
            
            while cur in hash_map:
                if hash_map[cur] != None:
                    cnt += hash_map[cur]
                    break
                cnt += 1
                cur -= 1
                
            hash_map[num] = cnt
            longest = max(longest, cnt)
            
        return longest

File: 128-longest-consecutive-sequence/128-longest-consecutive-sequence.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var longestConsecutive = function(nums) {
    if(nums.length <= 1) {
        return nums.length;
    }
    
    const set = new Set(nums);
    
    let longestLen = 0;
    
    nums.forEach(num => {
        
        if(!set.has(num) || set.has(num - 1)) {
            return;
        }
        
        let curLen = 1;
        set.delete(num);
        
        let nextConsecutiveNum = num + 1;
        
        while(set.has(nextConsecutiveNum)) {
            set.delete(nextConsecutiveNum);
            nextConsecutiveNum++;
            curLen++;
        }
        
        longestLen = Math.max(longestLen, curLen);
    });
    
    return longestLen;
};